README.md

```diff
@@ -89,33 +89,71 @@ token = OnlineToken(
 )
 ```
 
-#### Querying Shopify
+### Querying Shopify
 
-We can query the store using either the REST endpoint or the GraphQL endpoint:
+#### REST
+
+We can query the store using either the REST:
+
+```python
+token.execute_rest(
+  request: Request,
+  endpoint: str,
+  json: Optional[Dict[str, Any]],
+  debug: Optional[str],
+)
+```
+
+For example, if you want to query a product from shopify you can run:
+
+```python
+token.execute_rest(
+  request: GET,
+  endpoint: f'/products/{product_id}.json'
+)
+```
+
+If you want to update a product in a shop you can run:
 
 ```python
 token.execute_rest(
-  goodstatus,
-  method,
-  debug,
-  endpoint
+  request: PUT,
+  endpoint: f'/products/{product_id}.json',
+  json: {
+    "product":
+    {
+      "id": product_id,
+      "title": "New Title"
+    }
+  }
 )
+```
+
+The `REST` method takes a `request` parameter which is one of the `Request` constants defined in 
+the [rest](./spylib/utils/rest.py) file. The options are `GET`, `POST`, `PUT`, or `DELETE`.
+
+Endpoint is the API endpoint string that we are querying, this should be similar to 
+the following format:
+
+```python
+f'/{resource}.json?fields={resource.param}&{[params]}'
+```
+
+The `debug` parameter is the message that is returned when there is an error. It is optional as it defaults to `""`.
 
+#### GQL
```

Comment:
This also deserves the same details as REST such as type of the parameters and an example.